tempfile.go

```diff
@@ -173,3 +174,30 @@ func Symlink(oldname, newname string) error {
 
 	return os.RemoveAll(d)
 }
+
+// WriteFile mirrors ioutil.WriteFile, replacing an existing file with the same
+// name atomically.
+func WriteFile(filename string, data []byte, perm os.FileMode) error {
+	t, err := TempFile("", filename)
+	if err != nil {
+		return err
+	}
+	defer t.Cleanup()
+
+	// Set permissions before writing data, in case the data is sensitive.
+	if err := t.Chmod(perm); err != nil {
+		return err
+	}
+
+	n, err := t.Write(data)
+	if err != nil {
+		return err
+	}
+
+	// Fail if the write was short.
```

Comment:
From https://golang.org/pkg/os/#File.Write:

> Write writes len(b) bytes to the File. It returns the number of bytes written and an error, if any. Write returns a non-nil error when n != len(b).

Hence, I think we don’t need to check n explicitly, no?

Reply:
Done